pkg/reconciler/trigger/trigger.go

```diff
@@ -207,6 +240,37 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 	return nil
 }
 
+func (r *Reconciler) triggerMarkedReadyByDependency(dependencyObjRef corev1.ObjectReference, ctx context.Context, t *v1alpha1.Trigger) error {
+	gvk := schema.GroupVersionKind{
+		Group:   dependencyObjRef.GroupVersionKind().Group,
+		Version: dependencyObjRef.GroupVersionKind().Version,
+		Kind:    dependencyObjRef.Kind,
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	_, lister, err := r.buildInformerFactory.Get(gvr)
+	if err != nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting a lister for a resource with gvk '%+v', gvr '%+v'", gvk, gvr), zap.Error(err))
+		t.Status.MarkDependencyReadyUnknown("FailedToListResource", "Failed to list resource")
+		return err
+	}
+	dependencyObj, err := lister.ByNamespace(dependencyObjRef.Namespace).Get(dependencyObjRef.Name)
+	if err != nil {
+		logging.FromContext(ctx).Error("Unable to get the dependency", zap.Error(err))
+		if apierrs.IsNotFound(err) {
+			t.Status.MarkDependencyReadyUnknown("Dependency DoesNotExist", "Dependency does not exist")
+		} else {
+			t.Status.MarkDependencyReadyUnknown("DependencyGetFailed", "Failed to get dependency")
+		}
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("getting the dependency: %v", err)
```

And remove the log line on 258.